src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -146,7 +152,8 @@ public void testFromXContent() throws IOException {
     @Test
     public void testToQuery() throws IOException {
         QueryParseContext context = new QueryParseContext(index, queryParserService);
-        assertLuceneQuery(this.testQuery, this.testQuery.toQuery(context));
+        context.setAllowUnmappedFields(true);
```

Comment:
I am conflicted here: on one hand I don't like that we cover only one codepath (when the field doesn't exist), on the other hand I don't want to be testing mappers here too and how they generate queries. I think we can go ahead with unmapped fields at least for now.